### DISCUSSION TITLE
Precompile tasks

### DISCUSSION THREAD

**Discussion by jkthorne** [luckyframework/lucky#1361] [General]
Is there a way to have `lucky watch` or have a separate process in the `Procfile` precompile the tasks?

The tasks take a long time to run.  Even running the help command.  It also seems like the tasks are not cached to running the command `lucky` twice will recompile everything.  Is there a way to shorten this runtime?

**Comment by jwoertink** [comment]
We actually already check for precompiled tasks. https://github.com/luckyframework/lucky_cli/blob/master/src/lucky.cr#L18-L27 If you compile a task, and put it in to the `bin` directory named `lucky.whatever_it_is`, then lucky will use that version https://github.com/luckyframework/lucky_cli/blob/master/src/lucky.cr#L41-L43. You should have a `bin/lucky.watch` file.

Now, things like the help command can't be precompiled because in the case of `-h` it needs to know all of the tasks you have. If you pre-compiled the help task, then added a new custom task, it wouldn't be displayed. 

I do agree that some of these things take longer than what would be nice. I haven't checked, but maybe there's a bug in how that works? 🤔 Try switching out your web in Procfile.dev from `lucky watch` to `./bin/lucky.watch` and see if that runs faster. If it runs a ton faster, then we may need to look in to why that's not being picked up.
